### PR TITLE
Stop using the GitHub API to get PR information

### PR DESCRIPTION
**Pull request by foolip**

### CODE REVIEW

**Comment by foolip** [comment]
@Hexcles, the "Use the generalized get_pr_latencies for import stats" part of this PR is because with just the other changes, an assert began failing:
https://github.com/foolip/ecosystem-infra-stats/blob/a62c561e503a1fe9c7d160bcfd9b703c79cbb600/wpt-import-stats.py#L120

Output:
```
[306/1973] PR: https://github.com/w3c/web-platform-tests/pull/6429
PR merged at cd08181233218d4058da3f87df94f343964d1d2b imported at cd08181233218d4058da3f87df94f343964d1d2b
Chromium import 60b623808db8a5afb571db2b8325c156687c6bcf at 2017-09-15T14:39:58+00:00
Delay (mins): 419.283333333
[307/1973] PR: https://github.com/w3c/web-platform-tests/pull/6433
Traceback (most recent call last):
  File "wpt-import-stats.py", line 193, in <module>
    main()
  File "wpt-import-stats.py", line 188, in main
    latencies = get_latencies(imports, non_export_prs)
  File "wpt-import-stats.py", line 122, in get_latencies
    merge_commit, previous_import)
AssertionError: PR merge point 7a7d61501e55b291d73ad983b19678e7cd26ead3 before the previous import Import(cr_sha=u'f0e112939eebd94cc8af51216fa5ed367250239b', wpt_sha=u'170bf8330e41fd6ba0af816dd6fca4816cbf8d8d', date=u'2018-04-25T12:39:23+00:00')
```

That's because `index_found` was 0, and so `previous_import = imports[index_found - 1]` wraps around to the end of the list.

I'll first try to revert the import stats changes and just fix that bug.

**Comment by foolip** [comment]
There aren't very interesting changes to the output due to this change, just more PRs.

**Comment by foolip** [comment]
Changes can be seen in https://github.com/foolip/ecosystem-infra-stats/commit/f1e7d6660265058fa04a268f06afbde36c0ab478.

File: wpt-import-stats.py [hunk 1]
```diff
@@ -112,14 +112,15 @@ def get_latencies(imports, prs):
...
         assert _compare_commits(merge_commit, import_found.wpt_sha) <= 0, \
             "PR merge point {} after import {}".format(
             merge_commit, import_found)
-        assert _compare_commits(merge_commit, previous_import.wpt_sha) > 0, \
-            "PR merge point {} before the previous import {}".format(
-            merge_commit, previous_import)
+        if previous_import != None:
+            assert _compare_commits(merge_commit, previous_import.wpt_sha) > 0, \
+                "PR merge point {} before the previous import {}".format(
+                merge_commit, previous_import)
```
Comment (Hexcles) [inline wpt-import-stats.py]: This makes sense. Previously due to some coincidences (cutoff date, and API result pagination), it never happened.

File: wpt_common.py [hunk 2]
```diff
@@ -43,7 +38,7 @@
 
 def git(args, cwd):
     command = ['git'] + args
-    output = subprocess.check_output(command, cwd=cwd)
+    output = subprocess.check_output(command, cwd=cwd, env={'TZ': 'UTC'})
```
Comment (Hexcles) [inline wpt_common.py]: We might want to set the locale as well if we start to parse more git output.

File: wpt_common.py [hunk 3]
```diff
@@ -56,84 +51,64 @@ def wpt_git(args):
...
-def github_request(url):
-    base_url = 'https://api.github.com'
-    headers = None
-    if GH_TOKEN is not None:
-        headers = {'Authorization': 'token {}'.format(GH_TOKEN)}
-    res = requests.get(base_url + url, headers=headers)
-    res.raise_for_status()
-    return res.json()
+def verify_pr_tags(prs):
+    # sort PRs by commit date and assert that this matches commit DAG order
```
Comment (Hexcles) [inline wpt_common.py]: This is super awesome and gives me a lot of confidence!

File: wpt_common.py [hunk 4]
```diff
@@ -151,11 +126,10 @@ def pr_number_from_tag(tagish):
     return None
 
 
-def get_tagged_prs():
-    """Gets the set of integers for which merge_pr_* exists."""
+def get_merge_pr_tags():
+    """Gets the set of merge_pr_* tags as string."""
```
Comment (Hexcles) [inline wpt_common.py]: Well technically it's a list.

**Pull request merged**